server/src-lib/Hasura/Server/Logging.hs

```diff
@@ -221,14 +223,20 @@ mkHttpErrorLog userInfoM reqId req err query mTimeT compressTypeM =
            , olUserVars     = userVars <$> userInfoM
            , olResponseSize = respSize
            , olQueryExecutionTime = respTime
-           , olQuery = toJSON <$> query
+           , olQuery = toJSON <$> query'
+           , olRawQuery = toJSON <$> rawQuery
            , olError = Just $ toJSON err
            }
   in HttpLog L.LevelError $ HttpAccessLog http op
   where
     status = qeStatus err
     respSize = Just $ BL.length $ encode err
     respTime = computeTimeDiff mTimeT
+    query' = query >>= \q -> case q of
+      Object _ -> Just q
+      Array _  -> Just q
+      _        -> Nothing
```

Comment:
I think this is probably not the right approach, since it assumes all valid queries will always be objects or arrays, and it mixes concerns. I think the right solution is probably to change the type of the `query` argument to `Either ByteString Value`, or alternately, to make a new datatype `data SomeQuery = UnparsedQuery !ByteString | ParsedQuery !Value` (either is fine with me). Then the call site in `Hasura/RQL/Types/Column.hs` should be updated to pass the raw query using `Left`/`UnparsedQuery` without converting it to `Value` explicitly.